> Test looks real:
> 
> ```
> ======================================================================
> FAIL: corehq.apps.userreports.tests.test_pillow:ProcessRelatedDocTypePillowTest.test_process_doc_from_sql_stale
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/mnt/commcare-hq-ro/corehq/apps/userreports/tests/test_pillow.py", line 597, in test_process_doc_from_sql_stale
>     self._test_process_doc_from_sql_stale()
>   File "/mnt/commcare-hq-ro/corehq/apps/userreports/tests/test_pillow.py", line 612, in _test_process_doc_from_sql_stale
>     pillow.process_changes(since=since, forever=False)
>   File "/vendor/lib/python3.6/site-packages/django/test/testcases.py", line 85, in __exit__
>     '%d. %s' % (i, query['sql']) for i, query in enumerate(self.captured_queries, start=1)
> AssertionError: 16 != 4 : 16 queries executed, 4 expected
> ```
> 
> (Build has more debugging info.)

Fixed this by having the tests not checkpoint (which is what the extra queries were). I think that makes sense, and makes the tests less fragile, but let me know if you don't like that solution.

--- a/corehq/apps/userreports/tests/test_pillow.py
+++ b/corehq/apps/userreports/tests/test_pillow.py
@@ -612,9 +612,9 @@ class ProcessRelatedDocTypePillowTest(TestCase):
         http://manage.dimagi.com/default.asp?245341
         '''
         with mock.patch(
-                'corehq.pillows.case.KafkaCheckpointEventHandler.should_update_checkpoint'
-        ) as fake_checkpoint:
-            fake_checkpoint.return_value = False
+                'corehq.pillows.case.KafkaCheckpointEventHandler.should_update_checkpoint',
+                return_value=False
+        ):
             pillow = pillow or self.pillow
             for i in range(3):
                 since = pillow.get_change_feed().get_latest_offsets()
@@ -679,9 +679,9 @@ class ReuseEvaluationContextTest(TestCase):
 
     def _test_pillow(self, pillow, since, num_queries=12):
         with mock.patch(
-                'corehq.pillows.case.KafkaCheckpointEventHandler.should_update_checkpoint'
-        ) as fake_checkpoint:
-            fake_checkpoint.return_value = False
+                'corehq.pillows.case.KafkaCheckpointEventHandler.should_update_checkpoint',
+                return_value=False
+        ):
             with self.assertNumQueries(num_queries):
                 pillow.process_changes(since=since, forever=False)
 

